### PR TITLE
chore(mme): New unit tests for mme_app and code clean up

### PR DESCRIPTION
**Pull request by ulaskozat**
## Summary

- Unit test for failed session requests
- Unit test for network initiated session termination - single session
- Clean up unused procedure function

## Test Plan

`make test_oai` in dev VM.
run s1ap integ tests.

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.cpp [hunk 1]
```diff
@@ -164,28 +164,35 @@ void send_s6a_ula(const std::string& imsi, bool success) {
...
-  create_session_response_p->bearer_contexts_created.bearer_contexts[0]
-      .s1u_sgw_fteid.teid = 1000;
-  create_session_response_p->bearer_contexts_created.bearer_contexts[0]
-      .s1u_sgw_fteid.interface_type = S1_U_SGW_GTP_U;
-  create_session_response_p->bearer_contexts_created.bearer_contexts[0]
-      .s1u_sgw_fteid.ipv4 = 1;
-  create_session_response_p->bearer_contexts_created.bearer_contexts[0]
-      .s1u_sgw_fteid.ipv4_address.s_addr = 100;
+
+  create_session_response_p->teid              = 1;
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.cpp:173]: Can we replace these with macros?

File: lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.cpp [hunk 2]
```diff
@@ -164,28 +164,35 @@ void send_s6a_ula(const std::string& imsi, bool success) {
...
+  create_session_response_p->cause.cause_value = cause_value;
   create_session_response_p->bearer_contexts_created.bearer_contexts[0]
-      .eps_bearer_id                                                    = 5;
+      .cause.cause_value = cause_value;
   create_session_response_p->bearer_contexts_created.num_bearer_context = 1;
+
+  if (cause_value == REQUEST_ACCEPTED) {
+    create_session_response_p->paa.pdn_type            = IPv4;
+    create_session_response_p->paa.ipv4_address.s_addr = 1000;
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.cpp:181]: Can we use macros instead of hardcoded numbers?

**Review by ssanadhya — APPROVED** [review]
LGTM with minor comments

File: lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.cpp (hunk 2 above)
Comment (ulaskozat) [reply lte/gateway/c/core/oai/test/mme_app_task/mme_app_test_util.cpp:181]: I will pick these up in the follow up PR.

**Pull request merged**